src/v/coproc/event_handler.h

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2021 Vectorized, Inc.
+ *
+ * Licensed as a Redpanda Enterprise file under the Redpanda Community
+ * License (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * https://github.com/vectorizedio/redpanda/blob/master/licenses/rcl.md
+ */
+
+#pragma once
+
+#include "coproc/logger.h"
+#include "coproc/script_dispatcher.h"
+#include "coproc/wasm_event.h"
+#include "seastarx.h"
+
+#include <seastar/core/coroutine.hh>
+#include <seastar/core/future.hh>
+
+#include <absl/container/btree_map.h>
+#include <absl/container/btree_set.h>
+
+namespace coproc::wasm {
+
+class event_handler_exception final : public std::exception {
+public:
+    explicit event_handler_exception(ss::sstring msg) noexcept
+      : _msg(std::move(msg)) {}
+
+    const char* what() const noexcept final { return _msg.c_str(); }
+
+private:
+    ss::sstring _msg;
+};
+
+/// This class implements logic for handling events from internal coproc topic.
+/// There are 2 methods: for process event for specifix type and cron
+/// events(like heartbeats) We need store this handler to application, and pass
+/// it to register method of event_listener.
+class event_handler {
+public:
+    virtual ~event_handler() {}
+
+    virtual ss::future<> start() = 0;
+    virtual ss::future<> stop() = 0;
+
+    enum cron_finish_status { replay_topic, skip_pull, none };
+
+    /// Cron. Run it from loop in do_start event_listener function
+    virtual ss::future<event_handler::cron_finish_status> cron() = 0;
```

Comment:
I like the name `cron` but it might be easier for new users if it were named something more descriptive. for example, the async_event_hander::cron seems to process a heartbeat. should it be named `handle_event`? im just sure what its generic functionality is.

Reply:
`handle_event` is `process` method
event_listener runs `cron` before pulling event from `internl_coproc_topic`